packages/core/src/lib/base_protocol.ts

```diff
@@ -50,6 +56,52 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
+
+  protected async getPeers(peerId?: PeerId): Promise<Peer[]> {
+    const selectedPeers: Peer[] = [];
+
+    const { peer } = await selectPeerForProtocol(
+      this.peerStore,
+      [this.multicodec],
+      peerId,
+    );
+
+    selectedPeers.push(peer);
+
+    // only return multiple peers for protocols LP and Filter
+    if (
+      ![Codecs.LightPush, Codecs.FilterPush, Codecs.FilterSubscribe].includes(
+        this.multicodec,
+      )
+    )
+      return selectedPeers;
+
+    const allPeersForProtocol = await getPeersForProtocol(this.peerStore, [
+      this.multicodec,
+    ]);
+
+    // Counter to keep track of the number of bootstrap peers
+    let bootstrapPeerCount = 0;
+
+    allPeersForProtocol.map((peer) => {
+      // If the peer is a bootstrap peer and we don't have one yet, add it
```

Comment:
Would be good to have one comment that explain the overall intent of the logic here.

The correct logic seems to be :
if you only need 1 peer, return a bootstrap node
if you need more peer, return peer exhange peers